src/main/java/net/minecraftforge/client/player/inventory/tabs/GuiTabList.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package net.minecraftforge.client.player.inventory.tabs;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.FontRenderer;
+import net.minecraft.client.gui.GuiButton;
+import net.minecraft.client.gui.inventory.GuiContainer;
+import net.minecraft.client.renderer.RenderItem;
+import net.minecraft.client.resources.I18n;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.fml.client.config.GuiUtils;
+
+/**
+ * Enclosing class to keep GuiContainer clean. This handles all the survival tab drawing, and the addition of two buttons.
+ *
+ * @author Subaraki, Modified by Ash Indigo
+ */
+public class GuiTabList
+{
+    /**
+     * Dictates how many tabs per page are drawn
+     */
+    private int maxTabsPerPage = 12;
+    private int page;
+    /**
+     * Returns how many pages, a.k.a max button clicks to the right, can be done.
+     */
+    private int maxPages;
+    private GuiContainer guiContainer;
+    /**
+     * Cache used for comparison
+     */
+    private Class<? extends GuiContainer> guiContainerClass;
+    private List<GuiTab> tabs;
+    private static final ResourceLocation TEXTURE_TABS = new ResourceLocation("textures/gui/container/creative_inventory/tabs.png");
+
+    public GuiTabList(GuiContainer guiContainer)
+    {
+        this.guiContainer = guiContainer;
+        this.guiContainerClass = guiContainer.getClass();
+        tabs = new ArrayList<>(); // get a copy of the stored list so it can be modified without consequence
+        tabs.addAll(GuiTab.getTabsForGui(this.guiContainer.getClass()));
+        maxPages = setMaxPages();
+        getTabs();
+        for (GuiTab tab : getTabs())// Set the first page
```

Comment:
Is there any point to call `getTabs()` twice?

Reply:
Nope, noticed this as well.. removed